### PR TITLE
Convert cygwin paths to windows paths for java build tools

### LINKED ISSUE #154: Convert cygwin paths to windows paths for java build tools
Convert unix to windows class paths and hook tool resources files paths for java build tools.

### PR DESCRIPTION
**Pull request by vsebe**
Convert unix to windows class paths for java build tools.

fixes #154

Signed-off-by: Violeta Sebe <vsebe@ca.ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
Note an issue is not required unless you except or want to have some discussion before implementing a fix.

File: runtime/buildtools.mk [hunk 1]
```diff
@@ -89,6 +89,10 @@ findAllFiles = \
 	$(foreach i,$(wildcard $1/*),$(call findAllFiles,$i,$2)) \
 	$(wildcard $1/$2)
 
+# convert $1 Cygwin path to Windows path with regular slashes
+convertUnixToWinPath = \
+	$(if $(findstring CYGWIN,$(OS)), $(foreach f,$1,$(shell cygpath -m $f)),$1)
```
Comment (keithc-ca) [inline runtime/buildtools.mk]: cygpath can convert all paths in a single invocation: `foreach` is not needed.

File: runtime/buildtools.mk [hunk 2]
```diff
@@ -110,7 +114,7 @@ NLS_NLSTOOL := $(JAVA) -cp sourcetools/lib/j9nls.jar com.ibm.oti.NLSTool.J9NLS
 nls : buildtools
 	$(NLS_NLSTOOL) $(NLS_OPTIONS)
 
-HOOK_DEFINITION_FILES := $(strip $(call findAllFiles,.,*.hdf))
+HOOK_DEFINITION_FILES := $(call convertUnixToWinPath,$(strip $(call findAllFiles,.,*.hdf)))
```
Comment (keithc-ca) [inline runtime/buildtools.mk]: This shouldn't be necessary (nor have any effect): all the files should be of the form `./subdir/xxx.hdf`.

Comment (vsebe) [reply runtime/buildtools.mk]: just close it

Comment (vsebe) [reply runtime/buildtools.mk]: removed

File: sourcetools/buildj9tools.mk [hunk 3]
```diff
@@ -34,7 +34,8 @@ DEST_DIR    := lib
 EMPTY       :=
 LIB_DIR     := lib
 OS          := $(shell uname)
-PATH_SEP    := $(if $(findstring Windows,$(OS)),;,:)
+PATH_SEP    := $(if $(or $(findstring Windows,$(OS)),$(findstring CYGWIN,$(OS))),;,:)
+
```
Comment (keithc-ca) [inline sourcetools/buildj9tools.mk]: Can you remove this added (blank) line?

Comment (vsebe) [reply sourcetools/buildj9tools.mk]: done, thx

**Pull request merged**